src/jit/valuenum.cpp

```diff
@@ -256,87 +256,123 @@ TFp FpRem(TFp dividend, TFp divisor)
-        case GT_SUB:
-            return VNF_SUB_UN;
-        case GT_MUL:
-            return VNF_MUL_UN;
-
-        case GT_NOP:
-        case GT_COMMA:
-            return VNFunc(oper);
-        default:
-            unreached();
+        result = VNFunc(oper);
     }
-}
-
-ValueNumStore::ValueNumStore(Compiler* comp, CompAllocator alloc)
-    : m_pComp(comp)
-    , m_alloc(alloc)
-    , m_nextChunkBase(0)
-    , m_fixedPointMapSels(alloc, 8)
-    , m_checkedBoundVNs(alloc)
-    , m_chunks(alloc, 8)
-    , m_intCnsMap(nullptr)
-    , m_longCnsMap(nullptr)
-    , m_handleMap(nullptr)
-    , m_floatCnsMap(nullptr)
-    , m_doubleCnsMap(nullptr)
-    , m_byrefCnsMap(nullptr)
-    , m_VNFunc0Map(nullptr)
-    , m_VNFunc1Map(nullptr)
-    , m_VNFunc2Map(nullptr)
-    , m_VNFunc3Map(nullptr)
-    , m_VNFunc4Map(nullptr)
-#ifdef DEBUG
-    , m_numMapSels(0)
-#endif
-{
-    // We have no current allocation chunks.
-    for (unsigned i = 0; i < TYP_COUNT; i++)
+    else // We will need to use a VNF_ function
     {
-        for (unsigned j = CEA_None; j <= CEA_Count + MAX_LOOP_NUM; j++)
+        switch (oper)
         {
-            m_curAllocChunk[i][j] = NoChunk;
-        }
-    }
+            case GT_LT:
+                assert(unsignedOper);
+                assert(!overflowChecking);
+                result = VNF_LT_UN;
+                break;
 
-    for (unsigned i = 0; i < SmallIntConstNum; i++)
-    {
-        m_VNsForSmallIntConsts[i] = NoVN;
-    }
-    // We will reserve chunk 0 to hold some special constants, like the constant NULL, the "exception" value, and the
-    // "zero map."
-    Chunk* specialConstChunk = new (m_alloc) Chunk(m_alloc, &m_nextChunkBase, TYP_REF, CEA_Const, MAX_LOOP_NUM);
-    specialConstChunk->m_numUsed +=
-        SRC_NumSpecialRefConsts; // Implicitly allocate 0 ==> NULL, and 1 ==> Exception, 2 ==> ZeroMap.
-    ChunkNum cn = m_chunks.Push(specialConstChunk);
-    assert(cn == 0);
+            case GT_LE:
+                assert(unsignedOper);
+                assert(!overflowChecking);
+                result = VNF_LE_UN;
+                break;
 
-    m_mapSelectBudget = (int)JitConfig.JitVNMapSelBudget(); // We cast the unsigned DWORD to a signed int.
+            case GT_GE:
+                assert(unsignedOper);
+                assert(!overflowChecking);
+                result = VNF_GE_UN;
+                break;
 
-    // This value must be non-negative and non-zero, reset the value to DEFAULT_MAP_SELECT_BUDGET if it isn't.
-    if (m_mapSelectBudget <= 0)
-    {
-        m_mapSelectBudget = DEFAULT_MAP_SELECT_BUDGET;
+            case GT_GT:
+                assert(unsignedOper);
+                assert(!overflowChecking);
+                result = VNF_GT_UN;
+                break;
+
+            case GT_ADD:
+                if (overflowChecking)
+                {
+                    if (unsignedOper)
+                    {
+                        result = VNF_ADD_UN_OVF;
```

Comment:
Eh, I always wondered if the JIT shouldn't use distinct opers for overflow and unsigned. This would be one reason.